### ISSUE TITLE
[super_editor] - URLs should linkify when typed

### ISSUE THREAD

**Issue by matthew-carroll** [superlistapp/super_editor#746]
When users type URLs, they typically expect the URL to become a link automatically.

`SuperEditor` should recognize a URL, when typed. The URL should have a link attribution applied to it.

As the URL has characters added or removed, the attribution URL should be updated accordingly.

**Comment by darkCavalier11** [comment]
@matthew-carroll the expected behaviour, that is converting a URL to link is it like the Editor will now detect a pice of text and convert it to url if it suits the characteristics of a link
like typing `t.me/abc` should automatically parse it as a link and have link behaviour.

**Comment by matthew-carroll** [comment]
@darkCavalier11 I'm not sure I understand your comment. Can you rephrase it? Are you saying that this ticket should be closed? Or are you adding new details to the stated problem?

**Comment by darkCavalier11** [comment]
Yeah, i want to know more details to work on this if possibl
@matthew-carroll

**Comment by matthew-carroll** [comment]
This ticket is currently blocked by our work to switch to atomic commands. Atomic commands are going through a PR right now.

**Comment by darkCavalier11** [comment]
ok

**Comment by matthew-carroll** [comment]
@angelosilvestre Looking back at this ticket, I think this is done. Do you see anything in this ticket that we don't support now?

**Comment by angelosilvestre** [comment]
@matthew-carroll 

> As the URL has characters added or removed, the attribution URL should be updated accordingly.

It seems we are not doing this right now.

**Comment by angelosilvestre** [comment]
> @matthew-carroll
> 
> > As the URL has characters added or removed, the attribution URL should be updated accordingly.
> 
> It seems we are not doing this right now.

@matthew-carroll After looking at some other apps, I'm not sure if we should do this. Some apps update the URL, but most of them don't.

For example, the user might type "www.google.com", and then delete "www." and ".com", leaving just "google". We probably don't want to mess with the URL attribution.

**Comment by matthew-carroll** [comment]
So to be clear, you're suggesting a situation where someone can type www.google.com and then edit the text to be www.duckduckgo.com and it should still link to Google? If so, did you check that scenario in other apps, too? Is that how they work?

**Comment by angelosilvestre** [comment]
@matthew-carroll  Some keep the link to the original URL, some "unlinkify" when you type in a link and some update the URL. 

For example, google docs and notion do exactly what you described, obsidian updates the URL, and apple notes "unlinkifies".

Is more a question on how we want to handle this.

**Comment by matthew-carroll** [comment]
Should we make that an option for the reaction and implement all three within the reaction, depending on the chosen option?

**Comment by angelosilvestre** [comment]
Ok, we can do that.
